### PR TITLE
Move webcams loading to App and have Directions load Directions into App

### PR DESCRIPTION
**Pull request by kylebrain**

### CODE REVIEW

**Review by michaelmartinelli15 — DISMISSED** [review]

**Review by GarrettMoore22 — DISMISSED** [review]
Looks good!

File: src/App.tsx [hunk 1]
```diff
@@ -7,16 +7,40 @@ import TwitterScreen from "./components/TwitterScreen";
 import { MapScreen } from "./components/MapScreen";
 import { HashRouter, Switch, Route } from "react-router-dom";
 import "bootstrap/dist/css/bootstrap.min.css";
+import { CameraData, useCameras } from "./utils/useCameras";
 
 const App = (): JSX.Element => {
+    const [camData, isCamsLoading, camError] = useCameras();
+    const [directionResult, setDirectionResult] = useState<google.maps.DirectionsResult | null>(null);
```
Comment (ErylKenner) [inline src/App.tsx:14]: directionsResult, not directionResult (as per the google maps api)

File: src/App.tsx [hunk 2]
```diff
@@ -7,16 +7,40 @@ import TwitterScreen from "./components/TwitterScreen";
...
+        }
+
+        if (directionResult && !isCamsLoading) {
+            console.log("All loaded!");
+            console.log(directionResult);
+            console.log(camData);
+            // Call filter algorithm here
+            setfilteredCams(() => camData);
+        }
+    }, [directionResult, isCamsLoading, camError]);
```
Comment (ErylKenner) [inline src/App.tsx:30]: here

File: src/App.tsx [hunk 3]
```diff
@@ -7,16 +7,40 @@ import TwitterScreen from "./components/TwitterScreen";
...
+    const [directionResult, setDirectionResult] = useState<google.maps.DirectionsResult | null>(null);
+    const [filteredCams, setfilteredCams] = useState<CameraData[]>([]);
+
+    useEffect(() => {
+        if (camError) {
+            console.log("Failed to load cameras, cannot select camera along path. Encountered: " + camError);
+            return;
+        }
+
+        if (directionResult && !isCamsLoading) {
```
Comment (ErylKenner) [inline src/App.tsx:23]: here

File: src/App.tsx [hunk 4]
```diff
@@ -7,16 +7,40 @@ import TwitterScreen from "./components/TwitterScreen";
...
+            console.log("All loaded!");
+            console.log(directionResult);
```
Comment (ErylKenner) [inline src/App.tsx:25]: here

**Review by ErylKenner — CHANGES_REQUESTED** [review]

**Review by michaelmartinelli15 — APPROVED** [review]

**Review by ErylKenner — APPROVED** [review]

**Pull request merged**